cobbler/remote.py

```diff
@@ -1761,7 +1802,7 @@ def rename_menu(self, object_id: str, newname: str, token: Optional[str] = None)
         """
         return self.rename_item("menu", object_id, newname, token)
 
-    def new_item(self, what: str, token: str, is_subobject: bool = False):
+    def new_item(self, what: str, token: str, item_dict: dict = {}):
```

Comment:
This can't work since dicts are immutable. Thus the default needs to be `None`. This is not a problem since if the user passes nothing the argument is implicitly `None`.

However this means we need to handle the case that `item_dict` is `None`. After that we can then double check what the value of `is_subobject` is and added that to the `item_dict`.

Reply:
Yes you are right: `item_dict: Optional[Dict[str, Any]] = None`

Reply:
I have a clean solution with `**kwargs` that I will push later today. It even means we get to have a smaller diff for the PR... :D